canonical/rebac-admin
# WD-10181 - feature: add entitlements when creating a role

## Done

- Add entitlements panel.
- Submit the request to add entitlements once the role has been created.
- Update openapi.yaml to add the entitlements patch request.
- 

## QA

- Connect to the real API.
- Open the 'Create role' panel.
- Fill out the role name.
- Click "Add entitlements".
- Fill out the form. Some values need to be value so you can use something like: 
  - Resource type: "client"
  - Resource: "aws"
  - Entitlement: "can_view"
  - Click the "Create role" link at the top to go back to the first screen.
  - Click the "Create role" button and the role should get added.
  - If you want to see that the entitlements were created you can check the output of `fga tuple read --simple-output --api-url http://127.0.0.1:8080 --store-id $STORE_ID`


## Details

- https://warthogs.atlassian.net/browse/WD-10181

## Screenshots

<img width="807" alt="Screenshot 2024-04-11 at 3 52 36 PM" src="https://github.com/canonical/rebac-admin/assets/361637/0205fb37-252a-4793-a206-d4536f4d94dd">
